torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -370,6 +371,17 @@ def _get_preferred_device(self) -> torch.device:
         Return the preferred device to be used when creating tensors for collectives.
         This method takes into account the associated process group
         """
+        backend = dist.get_backend(self._process_group)
+        backend_config = dist.BackendConfig(dist.get_backend(self._process_group))
```

Comment:
nit: move this into the `else` branch since it is not used for NCCL case?

Reply:
and can reuse `backend` and not have to call `dist.get_backend` again?

Reply:
yes, i will fix it, i forgot to remove the old code as well.

Reply:
@awgu fixed the review comment.